src/core/interp.c

```diff
@@ -5543,6 +5543,10 @@ void MVM_interp_run(MVMThreadContext *tc, void (*initial_invoke)(MVMThreadContex
                 cur_op += 8;
                 goto NEXT;
             }
+            OP(uname):
+                MVM_platform_uname(tc, GET_REG(cur_op, 0).o);
+                cur_op += 2;
+                goto NEXT;
```

Comment:
I think I'd find this a bit more natural if we allocated a `BOOTStrArray`, populated it, and set it as the result, rather than relying on one being passed in; I don't think we'd really need the caller to pick the type here.

Reply:
Understood. I was cribbing from `getrusage`... there is a [commit](https://github.com/MoarVM/MoarVM/commit/fdb5e4d6bb15a14caa27bb35db8d751d4c24d131#diff-bb3587478c6b68f62c75192b00b05f0fR4543) that changes to passing in an array, so I cargo culted that through to this.

Reply:
That's a fair point. That particular op can be called repeatedly (as is by the scheduler), and I guess this was done to avoid regular allocations; I'm not that fond of the exception, but it's pragmatic. I don't think the information from `uname` changes, so there'd be little reason to repeatedly call it, so we can do things the usual way.